Update Git spec (#88)
postech-dao/simperby: docs/git.md

### Commit Format

### Branches

These are the names of the branches that are specially treated by the Simperby node. Branches other than `work` and `p` are managed by the node; it will be rejected if pushed.

> 1. `main`: always points to the last finalized block. It is strongly protected;user can't push to this branch.

Reviewer: ```suggestion
1. `main`: always points to the last finalized block. It is strongly protected; users can't push to this branch.
```

> 2. `work` the branch that the user can freely push or force-push. CLI commands like `create` interacts with this.

Reviewer: ```suggestion
2. `work`: the branch that users can freely push or force-push. CLI commands like `create` interacts with this.
```

> 2. `work`: the branch that users can freely push or force-push. CLI commands like `create` interacts with this.

Reviewer 2: ```suggestion
2. `work`: the branch that users can freely push or force-push. CLI commands like `create` interact with this.
```

3. `p`: the block proposal for this node. The node operator may push or force-push to this branch. When pushed, the Git server will check the validity of the branch. The consensus engine will recognize this branch and propose to the consensus. It stands for 'block proposal'.

> 4. `a-<number>`: a valid agenda (but not yet approved) propogated from other nodes. If the governance approved the agenda, it will point to the `agenda-proof` commit which lays on top of the agenda commit. The number is arbitrarily assigned.

Reviewer: ```suggestion
4. `a-<number>`: a valid agenda (but not yet approved) propagated from other nodes. If the governance has approved the agenda, it will point to the `agenda-proof` commit which lies on top of the agenda commit. The number is arbitrarily assigned.
```

> 5. `b-<number>`: a valid (but not yet finalized) block propogated from other nodes. The number is arbitrarily assigned.

Reviewer: ```suggestion
5. `b-<number>`: a valid (but not yet finalized) block propagated from other nodes. The number is arbitrarily assigned.
```

### Tags

> Tags can't be pushed by the user. Ther are always managed by the node.

Reviewer: ```suggestion
Tags can't be pushed by the users. They are always managed by the nodes.
```
